private/src/avb_streamhandler_app_socket/server.cpp

```diff
@@ -0,0 +1,1750 @@
+  {
+    requestedCmdStruct->signalFrequency = 997u;
+  }
+  if (0 == requestedCmdStruct->userParam)
+  {
+    switch(requestedCmdStruct->toneMode)
+    {
+    case IasAvbTestToneMode::eIasAvbTestToneSawtooth:
+      requestedCmdStruct->userParam = -1;
+      break;
+    case IasAvbTestToneMode::eIasAvbTestTonePulse:
+      requestedCmdStruct->userParam = 50;
+      break;
+    default:
+      break;
+    }
+  }
+
+  resultx = avbStreamHandler->setTestToneParams (requestedCmdStruct->localStreamId,
+                          requestedCmdStruct->channelIdx,
+                          requestedCmdStruct->signalFrequency,
+                          requestedCmdStruct->amplitude,
+                          requestedCmdStruct->toneMode,
+                          requestedCmdStruct->userParam
+                          );
+
+  std::cout << "\tResult: " << IasAvbResult::eIasAvbResultNotImplemented << std::endl;
+  AvbStreamHandlerSocketIpc::responseSocketIpc responseSocketIpcStruct;
+  responseSocketIpcStruct.command = requestedCmdStruct->command;
+  responseSocketIpcStruct.result = getResultString(resultx);
+
+  return responseSocketIpcStruct;
+}
+
+/// SuspendStreamhandler command
+AvbStreamHandlerSocketIpc::responseSocketIpc SuspendStreamhandler::execute (
+			IasMediaTransportAvb::IasAvbStreamHandler *avbStreamHandler,
+			AvbStreamHandlerSocketIpc::requestSocketIpc *requestedCmdStruct)
+{
+  //suppress temporarily
+  //IasMediaTransportAvb::IasAvbResult resultx = IasAvbResult::eIasAvbResultErr;
+
+  /* TODO: re-enable suspending avb streamhandler
+   Need to be implemented on server's main() first
+  resultx = avbStreamHandler->SuspendStreamhandler (requestedCmdStruct->suspendAction);
+  */
+  (void) *avbStreamHandler; //suppress temporarily
+  std::cout << "\tResult: " << IasAvbResult::eIasAvbResultNotImplemented << std::endl;
+  AvbStreamHandlerSocketIpc::responseSocketIpc responseSocketIpcStruct;
+  responseSocketIpcStruct.command = requestedCmdStruct->command;
+  responseSocketIpcStruct.result = IasAvbResult::eIasAvbResultNotImplemented;
+
+  return responseSocketIpcStruct;
+}
+
+namespace {
+
+#define FULL_VERSION_STRING "Version -P- " VERSION_STRING
+
+static const std::string cClassName = "Main::";
+#define LOG_PREFIX cClassName + __func__ + "(" + std::to_string(__LINE__) + "):"
+
+void handleSignal(int sig, siginfo_t *siginfo, void *context)
+{
+  (void)siginfo;
+  (void)context;
+
+  shutdownReason = sig;
+}
+
+void installSignals()
+{
+  struct sigaction act;
+  memset (&act, '\0', sizeof(act));
+  act.sa_sigaction = &handleSignal;
+  act.sa_flags = SA_SIGINFO;
+
+  (void)sigaction(SIGINT, &act, NULL);
+  (void)sigaction(SIGABRT, &act, NULL);
+  (void)sigaction(SIGTERM, &act, NULL);
+  (void)sigaction(SIGSEGV, &act, NULL);
+  (void)sigaction(SIGFPE, &act, NULL);
+  (void)sigaction(SIGUSR1, &act, NULL),
+  (void)sigaction(SIGUSR2, &act, NULL);
+
+}
+
+/*
+//TODO: Re-enable and update the suspending of avb streamhandler with signals.
```

Comment:
Do you know the state of this TODO?

Reply:
I don't think it is needed as main app will still be demo app and this will be used for smoke testing.